forge/forge/compile.py

```diff
@@ -267,6 +267,9 @@ def forge_compile_from_context(context: CompileContext) -> CompiledModel:
         CompileDepth.FINISH_COMPILE: finish_compile,
     }
 
+    if context.loss_module is not None:
+        context.compiler_cfg.enable_consteval = False
```

Comment:
What do you think about doing assert here, instead of implicit disable? 

I'm a bit scared for us to go into a domain where the compiler changes its own configurations. E.g. use will print config, see that cont eval is enabled, then during compile it'll report that we're skipping it.

Reply:
I agree that it can be confusing...

We can print a warning if you'd like that better...

Reply:
Also, we can drop this altogether and disable consteval manually when needed for now... But I would like to go this way, since I think it's better to automatically handle the failure cases.